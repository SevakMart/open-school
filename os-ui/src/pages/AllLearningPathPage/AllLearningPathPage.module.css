.mainContainer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin-top: 3rem;
}/*# sourceMappingURL=AllLearningPathPage.module.css.map */