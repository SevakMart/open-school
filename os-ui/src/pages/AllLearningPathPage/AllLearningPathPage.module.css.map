{"version":3,"sources":["AllLearningPathPage.module.scss","../../styles/Mixins/FlexMixins.scss","AllLearningPathPage.module.css"],"names":[],"mappings":"AAEA;ECDE,aAAA;EACA,mBDCqB;ECArB,mBAAA;EACA,8BDDgC;EAC9B,gBAAA;AEEJ","file":"AllLearningPathPage.module.css"}