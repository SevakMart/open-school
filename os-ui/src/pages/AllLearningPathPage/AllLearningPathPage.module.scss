@import "../../styles/Mixins/FlexMixins.scss";

.mainContainer{
    @include flexStyle(row,center,space-between);
    margin-top:3rem;
    .learningPathsMainContainer{
        border:2px solid red;
    }
}