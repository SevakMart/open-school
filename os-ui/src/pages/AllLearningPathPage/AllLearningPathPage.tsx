import { useMemo } from 'react';
import { userContext } from '../../contexts/Contexts';
import NavbarOnSignIn from '../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import MainContent from './Subcomponents/MainContent/MainContent';

const AllLearningPathPage = ({ userInfo }:{userInfo:object}) => {
  const idAndToken = useMemo(() => ({
    token: (userInfo as any).token,
    id: (userInfo as any).id,
  }), []);

  return (
    <>
      <NavbarOnSignIn currentUserEnrolled />
      <userContext.Provider value={idAndToken}>
        <MainContent />
      </userContext.Provider>
    </>
  );
};
export default AllLearningPathPage;
