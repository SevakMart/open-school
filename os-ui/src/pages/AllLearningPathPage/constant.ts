export const RATING = 'Rating';
export const DIFFICULTY = 'difficulty_id';
export const TITLE = 'Title';
