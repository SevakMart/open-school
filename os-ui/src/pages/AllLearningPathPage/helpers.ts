export const filterSendingParams = (sendingParams:object) => {
  const filteredParams = Object.entries(sendingParams).filter((params) => params[1].length > 0);
  return Object.fromEntries(filteredParams);
};
