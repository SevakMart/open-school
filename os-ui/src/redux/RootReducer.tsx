import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
