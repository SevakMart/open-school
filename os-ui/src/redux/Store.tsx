import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
