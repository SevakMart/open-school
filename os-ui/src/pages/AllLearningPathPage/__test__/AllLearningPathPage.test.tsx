import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { PayloadAction } from '@reduxjs/toolkit';
import { store } from '../../../redux/Store';
import AllLearningPathPage from '../AllLearningPathPage';

/* const mockUseNavigate = jest.fn();
const mockUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => mockUseLocation,
}));
jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () => () => (next: (action: PayloadAction) => void) => (action: PayloadAction) => next(action),
  initMessageListener: () => jest.fn(),
}));

describe('Create test cases for AllLearningPathPage component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <AllLearningPathPage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); */
