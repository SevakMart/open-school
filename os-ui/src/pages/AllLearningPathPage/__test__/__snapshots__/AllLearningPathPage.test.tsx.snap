// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for AllLearningPathPage component Create a snapshot test 1`] = `
<DocumentFragment>
  <nav
    class="mainContent"
  >
    <h2>
      string.homePage.navBar.logo
    </h2>
    <div
      class="navMainContent"
    >
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.allLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.myLearningPaths
      </p>
      <p
        class="navbarTitle"
      >
        string.homePage.navBar.mentors
      </p>
      <div
        class="bookmarkIconStyle"
      >
        <img
          alt="bookmark"
          src="bookMark.svg"
          style="font-size: 20px; cursor: pointer; fill: none;"
        />
      </div>
      <img
        alt="Notification"
        class="notificationIcon"
        src="Notification.svg"
      />
      <div
        class="userInfoContent"
      >
        <img
          alt="avatar"
          class="avatarLogo"
          src="https://reactjs.org/logo-og.png"
        />
        <img
          alt="Arrow down Icon"
          class="arrowDownIcon"
          src="Arrow-down.svg"
        />
      </div>
      <div />
    </div>
  </nav>
  <div
    class="mainContainer"
  >
    <div
      class="mainContainer"
      style="display: flex;"
    >
      <div
        class="filterMainContent"
      >
        <p
          class="mainTitle"
        >
          string.learningPath.filter
        </p>
      </div>
    </div>
    <div
      class="learningPathContainer"
    >
      <div
        class="headerMainContainer"
      >
        <div
          class="navAndSortContent"
        >
          <nav>
            <p
              class="activeNav"
              data-testid="All learning Path"
            >
              string.learningPath.all
            </p>
            <p
              class="nonActiveNav"
              data-testid="Saved Learning Path"
            >
              string.learningPath.saved
            </p>
          </nav>
          <div
            class="sortingContainer"
          >
            <label
              data-testid="sorting"
              for="sorting"
            >
              string.learningPath.sortBy
            </label>
            <select
              id="sorting"
              name="sorting"
            >
              <option
                value="Rating"
              >
                string.learningPath.rating
              </option>
              <option
                value="difficulty_id"
              >
                string.learningPath.difficulty
              </option>
              <option
                value="Title"
              >
                string.learningPath.title
              </option>
            </select>
          </div>
        </div>
        <div
          class="search search__allLearningPathSearch"
        >
          <div
            style="color: rgb(132, 138, 157); margin-left: 16px;"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
              />
            </svg>
          </div>
          <input
            data-testid="searchInput"
            placeholder="string.search"
            type="text"
          />
        </div>
      </div>
      <div
        class="mainCoreContainer"
      >
        <div
          class="mainContent"
        >
          <div
            class="loader"
          >
            <span />
            <span />
            <span />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
